Administration: Update site icon and media control button styling

Infinite-Null opened the pull request:
This PR updates the appearance of the Site Icon and media upload buttons to match their functionality. The previous styling made these elements look like drop areas, confusing when users attempted to drag and drop files onto them, as they only function as clickable buttons.

Trac ticket: [#47579](https://core.trac.wordpress.org/ticket/47579)

## Before
<img width="307" alt="Screenshot 2025-04-23 at 12 36 02 PM" src="https://github.com/user-attachments/assets/161ed255-c2d1-4521-aec2-36c92e9cb357" />

<img width="1262" alt="Screenshot 2025-04-23 at 12 35 39 PM" src="https://github.com/user-attachments/assets/a3ade381-9511-4f6b-98ed-6a97b46188dc" />

![image](https://github.com/user-attachments/assets/38a333ff-02b2-42f5-b020-6f2985e35196)


## After
![image](https://github.com/user-attachments/assets/958aa79c-251e-435e-8d45-688e718c4ec4)

<img width="1251" alt="Screenshot 2025-04-23 at 12 35 12 PM" src="https://github.com/user-attachments/assets/c7ab29ab-963a-4ab1-a17d-9a6e03b6ff61" />

<img width="299" alt="image" src="https://github.com/user-attachments/assets/6d4e405a-d32a-45a3-b66f-6da338251964" />

Infinite-Null committed:
Administration: Update site icon and media control button styling for clarity

Infinite-Null committed:
Administration: Update site icon and media control button styling for clarity

joedolson reviewed (changes requested):
These changes look good. Have you checked whether the `button-add-site-icon` class is still in use? If that was the only use of that class, then we should be able to remove the related CSS. 

((We can't remove `button-add-media`, as that's widely used by plugins and themes.)

I'd also like to see more of a compromise design: use the upload button styles, but keep the full width layout and padding.

Infinite-Null committed:
Update upload button and remove `button-add-site-icon` styles

Infinite-Null commented:
Hi @joedolson, Thank you for the feedback. I have made the necessary changes and updated the PR please review it at your convenience. 

### Screenshot:
![image](https://github.com/user-attachments/assets/7ef78c15-e2c2-4dfd-a9b6-8971ba91d17b)

joedolson commented:
Some additional changes that I think would help make things more consistent:

1) When an image is selected, e.g. for the background image, you end up with a small 'Remove' button and a large 'Change Image' button. I think it would be good if those were the same size, either both large or both small. I suggest small; I think that the Change Image button is different from the UPload Image button.

2) The Add Header image interface still uses the dashed outline design, but with a separate upload button. I think it would be good to make those all the same: eliminate the dashed placeholder space and use a large upload button.

These changes are otherwise good!

joedolson reviewed (changes requested):
Requested changes are noted in a comment; they're less changes to the existing code and more additions that would be helpful.

Infinite-Null committed:
Enhance header image control: update button classes for better styling and visibility

Infinite-Null committed:
Refactor: remove  button classes

Infinite-Null commented:
Hi @joedolson, I have done the requested changes. Please review it at your convenience:
### Upload Site Logo:
![image](https://github.com/user-attachments/assets/c0e8359b-662c-46d8-b3b2-4ed93ec647e6)

### After upload:
![image](https://github.com/user-attachments/assets/5fdd2bbd-302b-49df-8140-a3648f18bb1a)

### Header Upload:
![image](https://github.com/user-attachments/assets/8e6f496a-76cb-4774-b8a3-9ab2ab64e3c6)

### After Upload:
![image](https://github.com/user-attachments/assets/c507c0eb-ddc9-4a15-a1d1-1d9db4223295)

joedolson reviewed (changes requested):
By removing the class `upload-button`, this breaks the hook for the JS to trigger the selection events. So, for example, if you select a logo, the 'Change Logo' button no longer works.

I think what you need to do instead of removing the class is add a class to hook styling to when the selection is populated. E.g., add a class to the parent container like 'has-selection', then change the button style based on that parent class being present.

Infinite-Null committed:
Improves header image control UI

Infinite-Null committed:
Fixed focus state for upload button

Infinite-Null commented:
Hi @joedolson, Thank you for the feedback. I have updated the PR and fixed the issue. Also, I have attached a detailed testing video below. Please review it at your convenience. Thank you!

## Customize
https://github.com/user-attachments/assets/8163721f-711f-4546-a3f3-e8b82ff62b70

## General Settings
https://github.com/user-attachments/assets/59c71088-b3b4-4429-ad04-31889c9e6751

Infinite-Null committed:
Merge branch 'trunk' into fix/customizer-button-appearance

joedolson committed:
Set padding on mobile to match existing area

This makes the large buttons the same size as the inputs neighboring them, which better matches the current design on mobile.

joedolson reviewed (approved):
I made a minor change to improve the look on mobile, but otherwise I think this is good to go.

joedolson committed:
Add button-hero for site icon so styles match customizer

joedolson commented:
<img width="681" height="217" alt="Screenshot 2025-08-15 at 4 10 27 PM" src="https://github.com/user-attachments/assets/083a62e1-3e4a-4fb4-994e-260ae23da538" />

Changed the button size for General settings to `button-hero` for a closer match to existing styles.

shail-mehta reviewed (approved):
LGTM👍

shail-mehta commented:
**Customizer Settings:** 
|Before|After|
|-|-|
|<img width="300" height="300" alt="before-settings" src="https://github.com/user-attachments/assets/608fee97-ae05-4dce-9212-a799b593708b" /> | <img width="300" height="300" alt="after-settings" src="https://github.com/user-attachments/assets/f13a22ff-7713-42be-8478-5b3a875a893d" /> |

**Settings Icon:**
|Before|After|
|-|-|
|<img width="300" height="600" alt="before-customizer" src="https://github.com/user-attachments/assets/151d094f-d817-439e-9626-6d49f7623b64" /> | <img width="300" height="600" alt="after-customizer" src="https://github.com/user-attachments/assets/f776d750-36f4-412b-82e4-ac351383b2c9" /> |

joedolson committed:
Move header image button styles to customize-controls & reduce specificity

Infinite-Null commented:
Merged in changeset: [60645](https://core.trac.wordpress.org/changeset/60645)
